pkg/trtl/honu.go

```diff
@@ -35,17 +40,25 @@ func (h *HonuService) Get(ctx context.Context, in *pb.GetRequest) (*pb.GetReply,
 		log.Error().Msg("missing options in Trtl Get request")
 		return nil, status.Error(codes.InvalidArgument, "options must be provided in Get request")
 	}
-	log.Debug().Str("key", string(in.Key)).Bool("return_meta", in.Options.ReturnMeta).Msg("Trtl Get")
+
+	var key []byte
+	if len(in.Namespace) > 0 {
+		key = prepend(in.Namespace, in.Key)
+	} else {
+		key = in.Key
+	}
```

Comment:
Should we allow requests without namespaces? On the one hand, I can see a use case for adding keys without a prefix; however it would also mean that we'd have to allow the user to iterate over all keys in the database including the reserved namespaces. 

One way that we could avoid this is to have a `"default"` namespace that automatically gets applied if no namespace is provided. We would add `"default"` to our list of reserved namespaces and then automatically apply it in the case where no namespace is specified, making it optional. Alternatively we could simply require a namespace and return an invalid argument if no namespace is specified. 

Thoughts? We could open a new story for this if it requires more discussion.

Reply:
I figured we'd want to still have the option not to include namespaces?

Reply:
I think we have to answer the following questions:

1. Do we want users to have access to _all_ data in the database, including data from the reserved namespaces? If no, making namespace optional means having a `default` namespace, if yes, then namespace can be omitted.
2. Do we think of Trtl as a strict key/value store or something more like a document database (which is how GDS uses Trtl)? If the former, we should not require namespaces, if the latter, we should - it's a "collection" in Mongo terms.